### PR TITLE
Add the ACM TOMS paper to citation list

### PR DESCRIPTION
**Pull request by pratikvn**
This PR adds the recently published ACM TOMS paper to the CITING.md file. It additionally also adds a CITATION.cff file so that the github can nicely generate a tab to copy the citation.

### CODE REVIEW

File: CITING.md [hunk 1]
```diff
@@ -4,13 +4,22 @@ The main Ginkgo paper describing Ginkgo's purpose, design and interface is
...
 ``` bibtex
-@misc{anzt2020ginkgo,
-    title={Ginkgo: A Modern Linear Operator Algebra Framework for High Performance Computing},
-    author={Hartwig Anzt and Terry Cojean and Goran Flegar and Fritz Göbel and Thomas Grützmacher and Pratik Nayak and Tobias Ribizel and Yuhsiang Mike Tsai and Enrique S. Quintana-Ortí},
-    year={2020},
-    eprint={2006.16852},
-    archivePrefix={arXiv},
-    primaryClass={cs.MS}
+@article{ginkgo-toms-2022,
+title = {Ginkgo: {A} {Modern} {Linear} {Operator} {Algebra} {Framework} for {High} {Performance} {Computing}},
```
Comment (yhmtsai) [inline CITING.md:8]:
```suggestion
title = {{Ginkgo: A Modern Linear Operator Algebra Framework for High Performance Computing}},
```
should this also work?

**Review by MarcelKoch — COMMENTED** [review]
Perhaps you should also adjust the readme for that.

**Review by upsj — APPROVED** [review]
LGTM!

File: CITING.md [hunk 2]
```diff
@@ -4,13 +4,22 @@ The main Ginkgo paper describing Ginkgo's purpose, design and interface is
...
+url = {https://doi.org/10.1145/3480935},
+doi = {10/gphfsq},
+number = {1},
+urldate = {2022-02-17},
+journal = {ACM Transactions on Mathematical Software},
+author = {Anzt, Hartwig and Cojean, Terry and Flegar, Goran and Göbel, Fritz and Grützmacher, Thomas and Nayak, Pratik and Ribizel, Tobias and Tsai, Yuhsiang Mike and Quintana-Ortí, Enrique S.},
+month = feb,
+year = {2022},
+keywords = {ginkgo, healthy software lifecycle, High performance computing, multi-core and manycore architectures},
+pages = {2:1--2:33},
```
Comment (yhmtsai) [inline CITING.md:22]:
```suggestion
pages = {2:1--2:33}
```

File: README.md [hunk 3]
```diff
@@ -187,13 +187,22 @@ The main Ginkgo paper describing Ginkgo's purpose, design and interface is
...
+url = {https://doi.org/10.1145/3480935},
+doi = {10/gphfsq},
+number = {1},
+urldate = {2022-02-17},
+journal = {ACM Transactions on Mathematical Software},
+author = {Anzt, Hartwig and Cojean, Terry and Flegar, Goran and Göbel, Fritz and Grützmacher, Thomas and Nayak, Pratik and Ribizel, Tobias and Tsai, Yuhsiang Mike and Quintana-Ortí, Enrique S.},
+month = feb,
+year = {2022},
+keywords = {ginkgo, healthy software lifecycle, High performance computing, multi-core and manycore architectures},
+pages = {2:1--2:33},
```
Comment (yhmtsai) [inline README.md:205]:
```suggestion
pages = {2:1--2:33}
```

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  given-names: "Gregor"
+- family-names: "Ribizel"
+  given-names: "Tobias"
+- family-names: "Riemer"
+  given-names: "Lukas"
+- family-names: "Tsai"
+  given-names: "Yuhsiang"
+title: "Ginkgo: A Modern Linear Operator Algebra Framework for High Performance Computing"
+version: 1.4.0
+date-released:
```
Comment (yhmtsai) [inline CITATION.cff:32]: should it be the release date?

**Review by yhmtsai — APPROVED** [review]
LGTM. nit: the last , in bibtex should be removed

File: CITATION.cff (hunk 4 above)
Comment (pratikvn) [reply CITATION.cff]: Yes, I wanted to do that, but missed it. Thanks

**Review by MarcelKoch — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -187,13 +187,22 @@ The main Ginkgo paper describing Ginkgo's purpose, design and interface is
...
-    archivePrefix={arXiv},
-    primaryClass={cs.MS}
+@article{ginkgo-toms-2022,
+title = {{Ginkgo: A Modern Linear Operator Algebra Framework for High Performance Computing}},
+volume = {48},
+copyright = {All rights reserved},
+issn = {0098-3500},
+shorttitle = {Ginkgo},
+url = {https://doi.org/10.1145/3480935},
+doi = {10/gphfsq},
```
Comment (tcojean) [inline README.md:197]:
```suggestion
doi = {10.1145/3480935},
```

**Review by tcojean — APPROVED** [review]

**Pull request merged**